src/init.cpp

```diff
@@ -418,4 +420,5 @@ void DynamicDataLoader::check_consistency()
     mission_type::check_consistency();
     item_action_generator::generator().check_consistency();
     harvest_list::check_consistency();
+    profession::check_consistency_json_item_substitution();
```

Comment:
I suggest taking it a step further and defining the item_substitution class inside profession.cpp, nothing else has any business interacting with it, so it might as well only exist within that module.